src/FluentNHibernate/MappingModel/Collections/LayeredColumns.cs

```diff
@@ -27,16 +26,15 @@ public IEnumerable<ColumnMapping> Columns
         }
 
         public void AddColumn(int layer, ColumnMapping mapping)
-        {
+        {           
             if (!layeredValues.ContainsKey(layer))
-                layeredValues[layer] = new HashSet<ColumnMapping>(new ColumnMappingComparer());
-
-            ((HashSet<ColumnMapping>)layeredValues[layer]).Add(mapping);
+                layeredValues[layer] = new Dictionary<int, ColumnMapping>();

```

Comment:
A closer to the hash set implementation is:
`layeredValues[layer] = new Dictionary<ColumnMapping, ColumnMapping>();`
`((Dictionary<ColumnMapping, ColumnMapping>) layeredValues[layer]).Add(mapping, mapping);`